src/components/navigation/NavLink.tsx

```diff
@@ -0,0 +1,24 @@
+import { LinkProps } from "next/link";
+import Link from "next/link";
+import clsx from "clsx";
+import { ReactNode } from "react";
+
+interface NavLinkProps extends LinkProps {
+  children?: ReactNode;
+  name: string;
+  isActive?: boolean;
+  onClick?: () => void;
+}
+
+export default function NavLink(props: NavLinkProps) {
+  const linkClassName = clsx({
+    "font-semibold text-primary cursor-default": props.isActive,
+    "font-normal": !props.isActive,
+  });
+
+  return (
+    <Link className={linkClassName} href={props.href} onClick={props.onClick}>
+      {props.name}
+    </Link>
+  );
```

Comment:
You should use all `props` inside this component.

Also, use `children` property, not `name`

```suggestion
    <Link {...props} className={linkClassName} href={props.href} onClick={props.onClick}>
      {props.children}
    </Link>
  );
```